api/src/integration/java/org/pmiops/workbench/FireCloudIntegrationTest.java

```diff
@@ -29,46 +47,131 @@
+
+    // Get a refreshed access token for the FireCloud service account credentials.
+    serviceAccountCredential = serviceAccountCredential.createScoped(
+        FireCloudServiceImpl.FIRECLOUD_API_OAUTH_SCOPES);
+    serviceAccountCredential.refreshToken();
+  }
+
+  /**
+   * Creates a FireCloudService instance with the FireCloud base URL corresponding to the given
+   * WorkbenchConfig. Note that this will always use the test environment's default service account
+   * credentials when making API calls. It shouldn't be possible to make authenticated calls to the
+   * FireCloud prod environment.
+   * <p>
+   * This method mostly exists to allow us to run a status-check against both FC dev & prod within
+   * the same integration test run.
+   **/
+  private FireCloudService createService(WorkbenchConfig config) {
+    ApiClient apiClient = new ApiClient();
+    apiClient.setBasePath(config.firecloud.baseUrl);
+
+    apiClient.setAccessToken(serviceAccountCredential.getAccessToken());
+    apiClient.setDebugging(config.firecloud.debugEndpoints);
+
+    return new FireCloudServiceImpl(
         Providers.of(config),
         Providers.of(profileApi),
         Providers.of(billingApi),
-        // All of us groups api
-        Providers.of(groupsApi),
-        // End user groups api.
-        Providers.of(groupsApi),
+        Providers.of(allOfUsGroupsApi),
+        Providers.of(endUserGroupsApi),
         Providers.of(nihApi),
         Providers.of(workspacesApi),
-        Providers.of(new FireCloudConfig().statusApi(config)),
-        new FirecloudRetryHandler(new NoBackOffPolicy())
+        Providers.of(new StatusApi(apiClient)),
+        new FirecloudRetryHandler(new NoBackOffPolicy()),
+        serviceAccounts,
+        Providers.of(fireCloudAdminCredential)
     );
   }
 
-  @Test
-  public void testStatus() {
-    assertThat(fireCloudService.getFirecloudStatus()).isTrue();
-  }
-
-  private static WorkbenchConfig createConfig() throws Exception {
-    String testConfig = Resources.toString(Resources.getResource("config_prod.json"), Charset.defaultCharset());
+  private WorkbenchConfig loadConfig(String filename) throws Exception {
+    String testConfig = Resources.toString(Resources.getResource(filename), Charset.defaultCharset());
     WorkbenchConfig workbenchConfig = new Gson().fromJson(testConfig, WorkbenchConfig.class);
     workbenchConfig.firecloud.debugEndpoints = true;
     return workbenchConfig;
   }
 
+  private FireCloudService getTestService() throws Exception {
+    return createService(loadConfig("config_test.json"));
+  }
+
+  private FireCloudService getProdService() throws Exception {
+    return createService(loadConfig("config_prod.json"));
+  }
+
+  @Test
+  public void testStatusProd() throws Exception {
+    assertThat(getProdService().getFirecloudStatus()).isTrue();
+  }
+
+  @Test
+  public void testStatusDev() throws Exception {
```

Comment:
AFAICT, we only used to integration test against prod. This adds calls to both dev & prod. Unless there's a particular reason we don't want to test pinging both endpoints? If we find FireCloud dev is flaky at all, we can add a retry and/or remove the test.

Reply:
Let's keep an eye on this one for sure, to make sure that we aren't blocking releases to staging if FC dev is down.

Reply:
Good point. Maybe a more ideal solution would be for integration tests to support being parameterized by the deployment environment they're blocking, e.g. a staging deploy would run integration tests against FC prod.

But before that, we should develop a clearer strategy and definition on what our integration tests should cover. For instance, right now there's a warning at the top of these tests that says "Integration tests with real projects/clusters against production {workbench|notebooks} is not recommended at this time". I'm not quite sure that's enough to guide expansion of our integration-level coverage.